nbs/18_callback.fp16.ipynb

```diff
@@ -202,18 +202,49 @@
     "    def after_loss(self): self.autocast.__exit__(None, None, None)\n",
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #11.            div_factor:float=2., # Divide by this on overflow, multiply by this after scale_wait batches
</pre><p><code>test_overflow</code> is the one that directs this "on overflow"? (just curious)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/muellerzr/fastai-docment-sprint/pull/70/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='56'/>

Reply:
Yes if test_overflow finds the gradients have overflown, then we will divide the loss scale by this number. Otherwise we multiply by this number is we do not get an overflow after scale_wait number of batches.